### ISSUE TITLE
App does not compile with Ivy

### ISSUE THREAD

**Issue by ycatbink0t** [akveo/ng2-smart-table#1131]
Trace:
(node:3005) UnhandledPromiseRejectionWarning: Error: Error on worker #2: Error: Failed to compile entry-point ng2-smart-table (fesm5 as esm5) due to compilation errors:
node_modules/ng2-completer/src/modules/ng2-completer.module.d.ts:2:22 - error NG6002: Appears in the NgModule.imports of CellModule, but could not be resolved to an NgModule class

2 export declare class Ng2CompleterModule {
                       ~~~~~~~~~~~~~~~~~~
node_modules/ng2-completer/src/modules/ng2-completer.module.d.ts:2:22 - error NG6002: Appears in the NgModule.imports of FilterModule, but could not be resolved to an NgModule class

2 export declare class Ng2CompleterModule {
                       ~~~~~~~~~~~~~~~~~~
node_modules/ng2-smart-table/fesm5/ng2-smart-table.js:1003:5 - error NG6002: Appears in the NgModule.imports of TBodyModule, but itself has errors

1003 var CellModule = /** @class */ (function () {
         ~~~~~~~~~~
node_modules/ng2-smart-table/fesm5/ng2-smart-table.js:1433:5 - error NG6002: Appears in the NgModule.imports of THeadModule, but itself has errors

1433 var FilterModule = /** @class */ (function () {
         ~~~~~~~~~~~~
and so on

**Comment by arthurflor23** [comment]
Same issue

**Comment by vicenthy** [comment]
was already upgrading to angular 9

**Comment by BobCashStory** [comment]
same

**Comment by shadabk92** [comment]
whts the solution to this ?

**Comment by Nielsb85** [comment]
I am not sure yet. I think it is the way the package is build. I was trying
some stuff locally but did not had enough time yet to look in to it.

Op di 18 feb. 2020 06:25 schreef Shadab K <notifications@github.com>:

**Comment by custompro12** [comment]
> whts the solution to this ?

Unfortunately, it seems that the only solution at the moment is to [disable Ivy](https://angular.io/guide/ivy#opting-out-of-ivy-in-version-9) if your application depends on this library.

This library probably won't be updated to support Angular v9 as it is currently not really being maintained.  See [Low Maintenance](https://github.com/akveo/ng2-smart-table/commit/fa801f063b68cbfaa5ae163a55fb5308c44ccca9) notice in the README

**Comment by BobCashStory** [comment]
@custompro12 not a real solution since ivy make the build way more faster and that why we want to upgrade

**Comment by arthurflor23** [comment]
I think this issue is more angular or angular/cli [here](https://github.com/angular/angular/issues/35536)

**Comment by giansalex** [comment]
First `ng2-completer` should be updated.

https://github.com/oferh/ng2-completer/issues/443

**Comment by dmantelli** [comment]
Same! It's the only component in my project that isn't compatible with Ivy :(

**Comment by giansalex** [comment]
Other table component options?

**Comment by mohsenmahoski** [comment]
is there any update or any solution for compatible with Ivy?

**Comment by osamaalnuimi** [comment]
i think the problem from ng2-completer 
try to install this version 
npm install --ignore-scripts ng2-completer@next

**Comment by giansalex** [comment]
After install `ng9` and `ng2-completer@next`, this change is required:
![image](https://user-images.githubusercontent.com/14926587/76474575-2ac02300-63ca-11ea-8f32-4907dec91790.png)

**Comment by majhoos** [comment]
Can we have a time frame on this ng2-smart-table issue? If this component is just an experimental toy-ish component with no potential future then I personally need to hunt for some other components. I am very much surprised that this library with almost 15K downloads a week is getting this level of disregard and abandonment by its own development team.

**Comment by lenichols** [comment]
@majhoos +1

**Comment by HackPoint** [comment]
@majhoos I totally agree with you, but I believe they have this product for money, so the have separated the commercial version and this one. In this one they don't invest time.

A question for you, have you found a replacement for this component?

**Comment by lenichols** [comment]
@HackPoint where is the paid version? If its updated and working I'll need to subscribe to it.

**Comment by majhoos** [comment]
@HackPoint I have not made any final decision on this yet but one option I have considered so far is the community edition of ag-gird which is free. Their enterprise version is quite expensive and restrictive (in terms of installation) which makes it economically unviable.

**Comment by evereq** [comment]
Maybe at least merge https://github.com/akveo/ng2-smart-table/pull/1319, which seems to be approved already... We will wait a bit more time, but in the worse case, we will fork and update ourselves (using Smart Table in https://github.com/ever-co/ever-gauzy heavily, not easy to replace)
